action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -282,22 +282,25 @@ func (stateDB *StateDBAdapter) GetNonce(evmAddr common.Address) uint64 {
 		log.L().Error("Failed to convert evm address.", zap.Error(err))
 		return 0
 	}
-	nonce := uint64(0)
+	pendingNonce := uint64(1)
 	state, err := stateDB.AccountState(addr.String())
 	if err != nil {
 		log.L().Error("Failed to get nonce.", zap.Error(err))
 		// stateDB.logError(err)
 	} else {
-		nonce = state.Nonce
+		pendingNonce = state.PendingNonce()
 	}
-	if stateDB.usePendingNonce {
-		nonce++
+	if !stateDB.usePendingNonce {
+		if pendingNonce == 0 {
+			panic("invalid pending nonce")
+		}
+		pendingNonce--
```

Comment:
why need `--` here? since it's using `pendingNonce = state.PendingNonce()` above?
at least for old type account, say `state.Nonce = 4`, it should return 5

Reply:
line 293, `if !stateDB.usingPendingNonce {`